### PR TITLE
Add flow type checking to Icon component

### PR DESCRIPTION
**Pull request by jasononeil**
Most of this work was done by @nikuda in #110, I'm splitting it out into a smaller PR.

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -12,6 +12,6 @@
 
 [options]
 module.name_mapper='.*\.s?css$' -> 'css-module-flow'
-module.name_mapper.extension='svg' -> 'empty/object'
+module.name_mapper.extension='svg' -> '<PROJECT_ROOT>/util/svg-loader-stub'
```
Comment (sentience) [inline .flowconfig]: Niiiice.

File: components/Icon/Icon.js [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import styles from './Icon.module.scss';
 import { warn } from '../../util/error';
 import { enableUniqueIds } from 'react-html-id';
 
+export type IconType = {
+  id: string,
+  viewBox: string,
+};
```
Comment (sentience) [inline components/Icon/Icon.js]: I think we call a similar type `SvgAsset` in Elm. Would that be a clearer name to describe what this is?

Comment (jasononeil) [reply components/Icon/Icon.js]: Yes it would. I'll rename :)

**Pull request merged**